mailpoet/lib/Automation/Integrations/WordPress/Fields/CommentFieldsFactory.php

```diff
@@ -0,0 +1,170 @@
+<?php declare(strict_types = 1);
+
+namespace MailPoet\Automation\Integrations\WordPress\Fields;
+
+use MailPoet\Automation\Engine\Data\Field;
+use MailPoet\Automation\Engine\WordPress;
+use MailPoet\Automation\Integrations\WordPress\Payloads\CommentPayload;
+
+class CommentFieldsFactory {
+
+  /** @var WordPress */
+  private $wp;
+
+  public function __construct(
+    WordPress $wp
+  ) {
+    $this->wp = $wp;
+  }
+
+  /**
+   * @return Field[]
+   */
+  public function getFields(): array {
+    return [
+      new Field(
+        'wordpress:comment:id',
+        Field::TYPE_INTEGER,
+        __('Comment ID', 'mailpoet'),
+        function (CommentPayload $payload) {
+          return $payload->getCommentId();
+        }
+      ),
+      new Field(
+        'wordpress:comment:author-name',
+        Field::TYPE_STRING,
+        __('Comment author name', 'mailpoet'),
+        function (CommentPayload $payload) {
+          $comment = $payload->getComment();
+          //phpcs:ignore Squiz.NamingConventions.ValidVariableName.MemberNotCamelCaps
+          return $comment ? $comment->comment_author : '';
```

Comment:
Same question here — are the empty strings just for PHPStan, or should they be `null`?

Reply:
https://github.com/mailpoet/mailpoet/pull/5197/commits/a20cc87ee8322182010d9eb7f38cee78b7814352